apiplugin/file/filehost.go

```diff
@@ -50,10 +51,12 @@ type FilePlugin struct{}
 type fileInfo struct {
 	id             file.FileId
 	path           string
+	length         int // length of the file content
 	content        string
 	status         FileStatus
-	createDate     time.Time
-	lastAccessTime time.Time
+	createDate     string
+	lastAccessTime string
+	prevRune       int // index of previous rune; or -1
```

Comment:
The question of whether to use runes or bytes is hard to say.  I can see both being useful.  For now, you just need to be consistent in the API.  If you are going to use runes then the "customer" of your API should be passing in a slice of runes `[]rune`.  If you want to use bytes (which is probably easier) just be sure everything is bytes (and not runes).  Runes can be larger than a byte, so you probably want insure that your code that reads/writes data is using an array of runes, at least before the read or write is called.

FWIW, the go team has decided to stay with bytes and provide utilities to convert []byte to []rune and vice version... that's why the Reader and Writer interfaces use bytes.